core/controllers/reader_test.py

```diff
@@ -115,41 +116,47 @@ def setUp(self):
         self.enable_string_classifier = self.swap(
             feconf, 'ENABLE_STRING_CLASSIFIER', True)
 
-        # Reading YAML exploration into a dictionary.
-        yaml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                 '../tests/data/string_classifier_test.yaml')
-        with open(yaml_path, 'r') as yaml_file:
-            self.yaml_content = yaml_file.read()
-
         self.login(self.VIEWER_EMAIL)
         self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
 
         # Load demo exploration.
-        self.exp_id = '0'
+        self.unused_exp_id = '0'
+        self.exp_id = '16'
         self.title = 'Testing String Classifier'
         self.category = 'Test'
-        exp_services.delete_demo(self.exp_id)
-        exp_services.load_demo(self.exp_id)
-
-        # Creating the exploration domain object.
-        self.exploration = exp_domain.Exploration.from_untitled_yaml(
-            self.exp_id,
-            self.title,
-            self.category,
-            self.yaml_content)
+        self.state_name = 'Home'
+        exp_services.delete_demo(self.unused_exp_id)
+        exp_services.load_demo(self.unused_exp_id)
+
+        test_exp_filepath = os.path.join(
+            feconf.TESTS_DATA_DIR, 'string_classifier_test.yaml')
+        yaml_content = utils.get_file_contents(test_exp_filepath)
```

Comment:
To load the exploration using exploration id and version in classifiy_string_classifier_rule, the exploration needs to be saved in data. The classifier demo exploration can be loaded from yaml, but it cannot be retrieved by exp_id. So, I saved the classifier demo exploration with an id of 16 so that it can be later retrieved by used get_exploration_by_id(). I followed the procedure in classifier_services_test.